File: StringBase.cs
using System;

public class StringBase
{
    protected string text;

    public StringBase()
    {
        text = "Hello";
    }

    public StringBase(string str)
    {
        text = str;
    }

    public StringBase(StringBase other)
    {
        text = other.text;
    }

    public int GetLength()
    {
        return text.Length;
    }

    public string GetText()
    {
        return text;
    }
}

File: Lab2.1/StringBase.cpp
#include "StringBase.h"

StringBase::StringBase() : text("Hello") {}

StringBase::StringBase(const std::string& str) : text(str) {}

StringBase::StringBase(const StringBase& other) : text(other.text) {}

int StringBase::getLength() const {
    return text.length();
}

std::string StringBase::getText() const {
    return text;
}
